generator/template.go

```diff
@@ -229,20 +329,38 @@ func fieldName(r *registry.Registry) func(name string) string {
 func renderURL(r *registry.Registry) func(method data.Method) string {
 	fieldNameFn := fieldName(r)
 	return func(method data.Method) string {
-		url := method.URL
+		methodURL := method.URL
 		reg := regexp.MustCompile("{([^}]+)}")
-		matches := reg.FindAllStringSubmatch(url, -1)
+		matches := reg.FindAllStringSubmatch(methodURL, -1)
+		fieldsInPath := "["
 		if len(matches) > 0 {
 			log.Debugf("url matches %v", matches)
 			for _, m := range matches {
 				expToReplace := m[0]
 				fieldName := fieldNameFn(m[1])
 				part := fmt.Sprintf(`${req["%s"]}`, fieldName)
-				url = strings.ReplaceAll(url, expToReplace, part)
+				methodURL = strings.ReplaceAll(methodURL, expToReplace, part)
+				fieldsInPath += fmt.Sprintf(`"%s", `, fieldName)
+			}
+		}
+		fieldsInPath = strings.TrimRight(fieldsInPath, ", ") + "]"
+
+		if !method.ClientStreaming && method.HTTPMethod == "GET" {
+			// parse the url to check for query string
+			parsedURL, err := url.Parse(methodURL)
+			if err != nil {
+				return methodURL
+			}
+			renderURLSearchParamsFn := fmt.Sprintf("${fm.renderURLSearchParams(req, %s)}", fieldsInPath)
+			// prepend "&" if query string is present otherwise prepend "?"
+			if parsedURL.RawQuery != "" {
+				methodURL += "&" + renderURLSearchParamsFn
```

Comment:
you might want to remove the ending & if there's any before attaching

Reply:
say if the url in the option is `/paht/to/{a}?test=a&`, then you don't need to add the `&`

Reply:
Ah indeed! Thank you for pointing that out 🙇🏼

Reply:
I was trying to write a test case for this and `grpc-gateway` throws an error when trying to generate.
`--grpc-gateway_out: unexpected token "&test=1" after segments "api/query/{a}": /api/query/{a}&test=1`
It seems we are not allowed to inject query parameters in the path and therefore I don't think we need the parse the method URL to check for query parameters.
Let me know what you think about this 🙏🏼

Reply:
this is new to me. looks like an issue on the grpc-gateway. let's leave the logic here coz I don't see there's anything  in the `http.proto` saying it's not permitted?

Reply:
Ok sure. I will make the modifications as per your suggestion 👍🏼

Reply:
hangon. found a problem in  `/api/query/{a}&test=a` shouldn't this be a `?` to start with

Reply:
should be `/api/query/{a}?test=a`

Reply:
Sorry for the typo in my comment, this is the exact error message I got:
`--grpc-gateway_out: unexpected token "?test=a" after segments "api/query/{a}": /api/query/{a}?test=a`

Reply:
![Screenshot from 2021-04-27 13-31-41](https://user-images.githubusercontent.com/44151328/116185628-f427a200-a75c-11eb-99fd-0afa27fa9cb4.png)
Leaving a screenshot for clarification 🙇🏼

Reply:
Have modified the function as per your suggestion to remove leading `&` if present 🙇🏼